server.go

```diff
@@ -0,0 +1,207 @@
+func ErrorRespond(w http.ResponseWriter, errMsg string, code int) {
+	resp := &ErrorResponse{Error: errMsg}
+
+	w.Header().Set(contentType, jsonContentType)
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(code)
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		log.Errorf("Error occured when marshalling response: %s", err)
+	}
+}
+
+type CreateEmailAccountResponse struct {
+	Id string `json:"id"`
+}
+
+func CreateEmailAccountHandler(db *sql.DB) func(w http.ResponseWriter, req *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		newAccount := &EmailAccount{}
+		body, err := readReqBody(r)
+		if err != nil {
+			ErrorRespond(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if err := json.Unmarshal(body, newAccount); err != nil {
+			log.Errorf("Error occured when unmarshalling data: %s", err)
+			ErrorRespond(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		err = newAccount.Save(db)
+		if err != nil {
+			ErrorRespond(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		resp := &CreateEmailAccountResponse{
+			Id: newAccount.Id,
+		}
+
+		w.Header().Set(contentType, jsonContentType)
+		w.WriteHeader(http.StatusCreated)
+		if err := json.NewEncoder(w).Encode(resp); err != nil {
+			log.Errorf("Error occured when marshalling response: %s", err)
+			return
+		}
+	}
+}
+
+type SendEmailRequest struct {
+	EmailData  EmailData `json:"email_data"`
+	SecureOnly bool      `json:"secure_only,omitempty"`
+}
+
+func SendEmailHandler(db *sql.DB, emailPool *emailLib.Pool) func(w http.ResponseWriter, req *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		id := mux.Vars(r)["id"]
+
+		sendEmailReq := &SendEmailRequest{}
+		body, err := readReqBody(r)
+		if err != nil {
+			ErrorRespond(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if err := json.Unmarshal(body, sendEmailReq); err != nil {
+			log.Errorf("Error occured when unmarshalling data: %s", err)
+			ErrorRespond(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		// TODO - support returning 500 as well
+		emailAccount, err := GetEmailAccount(id, db)
+		if err != nil {
+			ErrorRespond(w, err.Error(), http.StatusNotFound)
+			return
+		}
+
+		if sendEmailReq.SecureOnly && !emailAccount.HasPubKey {
+			log.Warnf("Failed SecureOnly Email to %s - no pub key", emailAccount.Id)
+			ErrorRespond(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		err = emailAccount.Send(sendEmailReq.EmailData, emailPool)
+		if err != nil {
+			log.Errorf("Error sending email: %v", err)
+			ErrorRespond(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusNoContent)
+	}
+}
+
+type SendBulkEmailRequest struct {
+	Ids        []string  `json:"ids"`
+	EmailData  EmailData `json:"email_data"`
+	SecureOnly bool      `json:"secure_only,omitempty"`
+}
+
+type SendBulkEmailResponse struct {
+	FailedEmails []string `json:"failed_emails"`
```

Comment:
@ajvb Rename to `FailedIds` or `FailedEmailIds` or `FailedAccountIds`; we'll probably add a feature where you can send an email to someone by email(!) rather than by ID.